plugins/search/src/components/SearchPage/SearchPage.tsx

```diff
@@ -91,9 +93,15 @@ export const UrlUpdater = () => {
  */
 export const SearchPage = () => {
   const outlet = useOutlet();
+  const configApi = useApi(configApiRef);
+
+  const initialState = {
+    ...getSearchContextInitialStateDefaults(),
+    pageLimit: configApi.getOptionalNumber('app.search.initialState.pageLimit'),
```

Comment:
I wonder if we could read the configuration inside the SearchContextProvider, that way the context state will know how to initialize itself in an encapsulated way, independent of where the context is rendered... As a result, we do not need to read the configuration repeatedly in the search page and modal components, or in any other custom search component that consumes the search context). How do you feel about this suggestion?